### DISCUSSION TITLE
How to log username in access log?

### DISCUSSION THREAD

**Discussion by defensivedepth** [ory/kratos#2208] [Q&A] [answered]
Is there a way to log traits (ie human-readable username) along with the identity id? 

`{"http_request":{"headers":{"accept":"application/json, text/plain, */*","accept-encoding":"gzip, deflate, br","accept-language":"en-US,en;q=0.5","connection":"close","cookie":"Value is sensitive and has been redacted. To see the value set config key \"log.leak_sensitive_values = true\" or environment variable \"LOG_LEAK_SENSITIVE_VALUES=true\".","referer":"https://*****/","sec-fetch-dest":"empty","sec-fetch-mode":"cors","sec-fetch-site":"same-origin","user-agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:96.0) Gecko/20100101 Firefox/96.0","x-forwarded-for":"******","x-forwarded-proto":"https","x-real-ip":"******"},"host":"******","method":"GET","path":"/sessions/whoami","query":null,"remote":"1****","scheme":"http"},"http_response":{"headers":{"content-type":"application/json; charset=utf-8","vary":"Cookie","x-kratos-authenticated-identity-id":"61c0972b-3ced-4b24-9bdf-3c47fa4c62a9"},"size":776,"status":200,"text_status":"OK","took":162366},"level":"info","msg":"completed handling request","time":"2022-02-10T20:09:06Z"}
`

**Comment by defensivedepth** [comment, marked as answer]
Bump, anyone have any ideas?

**Reply by Benehiko** [reply]
Hi @defensivedepth 

Not sure if this is possible. But maybe you could try set the log level to `debug` instead of `info`.

**Reply by vinckr** [reply]
Hey @defensivedepth , just checking in if your problem still persists or if setting log level to `debug` solved it.

**Reply by defensivedepth** [reply]
Enabling `debug` logs in production is not a viable longterm solution.

I filed a feature request here: https://github.com/ory/kratos/issues/2308

Thanks!
